src/components/adminTenant/tenantSpaces.js

```diff
@@ -0,0 +1,73 @@
+import React, { useState, useEffect } from 'react';
+import { Tabs, Tab, Container } from 'react-bootstrap';
+import { useHistory } from 'react-router-dom';
+import Spaces from '../viewSpaces/Spaces';
+import {getTenantRegisteredSpaces} from "../../utils/HTTPrequests";
+import moment from 'moment';
+import queryString from 'query-string';
+
+const TenantSpaces = () => {
+  const [key, setKey] = useState("current")
+  const [spaces, setSpaces] = useState([])
+  const [error, setError] = useState("")
+  const [moreInfoDisplay, setMoreInfoDisplay] = useState(false)
+  const [spaceId, setSpaceId] = useState("")
+
+  const history = useHistory();  
+
+  const infoFunction = (spaceId) => {
+    return ()  => window.open(`/space?_id=${spaceId}&startDate=${moment().format("YYYY-MM-DD")}&endDate=${moment().add(1,'days').format("YYYY-MM-DD")}`)
+  }
+
+  const fetchSpacesData = async() =>{
+    try{
+      const userSpaces = await getTenantRegisteredSpaces(`?state=${key}`) || []
+      console.log(userSpaces)
+      setSpaces(userSpaces)
+    }catch(err) {
+      setError(err)
+    }
+  }
+
+  useEffect(() => {
+    fetchSpacesData()
+  },[]) 
+
```

Comment:
why this useEffect? the same function will be executed the first time in the second use effect.